src/lib/model/stateMachine/states/invokeRetry.ts

```diff
@@ -0,0 +1,88 @@
+import { assign, createMachine } from 'xstate';
+import SDKStandardComponents from '@mojaloop/sdk-standard-components';
+import Logger = SDKStandardComponents.Logger.Logger;
+// import { StateMachine } from 'xstate/lib/types';
+
+type AsyncFunc = (ctx?: any) => Promise<any>;
+
+interface InvokeRetryOpts {
+  id: string;
+  // service: AsyncFunc | StateMachine<any, any, any>;
+  service: AsyncFunc;
+  maxRetries?: number;
```

Comment:
Should we not have defaults for `maxRetries` and r`etryInterval`?

I see `invokeRetry` being used below, but I don't see those Opts being configured if they are not being configured...whats the reason for using it?

Reply:
I was planning to make this module more generic, so it can be part of other projects as well.
For mgmt API we don't use `maxRetries` (always retry).
`retryInterval` can be configured in future if required.